<a href="https://colab.research.google.com/github/anjalixprakash/Project2/blob/main/Project_2_Part_1_(Core).ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# Project 2 - Part 1 (core)

Anjali Prakash

Source 1: https://www.kaggle.com/datasets/swathiunnikrishnan/amazon-consumer-behaviour-dataset

Source 2: https://www.kaggle.com/datasets/wenruliu/adult-income-dataset


In [14]:
from google.colab import drive
drive.mount('/content/drive')

Drive already mounted at /content/drive; to attempt to forcibly remount, call drive.mount("/content/drive", force_remount=True).


In [15]:
import pandas as pd

In [19]:
# Load data
amazondata = '/content/drive/MyDrive/CodingDojo/02-MachineLearning/Week07/Data/Amazon Customer Behavior Survey.csv.xls'
df = pd.read_csv(amazondata)
df.head()


(602, 23)

First choice: dataset 1

1. Source of data
- Kaggle

2. Brief description of data
- This data provides Amazon consumer insights
3. What is the target?
- I'm setting the target to be 'customer_satisfaction'.

4. What does one row represent? (A person? A business? An event? A product?)
- A single row represents a person

5. Is this a classification or regression problem?
- This would be a classification problem
6. How many features does the data have?
- 23
7. How many rows are in the dataset?
- 602 rows
8. What, if any, challenges do you foresee in cleaning, exploring, or modeling this dataset?
- Cleaning the timestamp column if I plan to use it. Shortening some of the object columns while still maintaining it's relavance for easier readability.

In [20]:
incomedata = '/content/drive/MyDrive/CodingDojo/02-MachineLearning/Week07/Data/adult.csv'
dff = pd.read_csv(incomedata)
dff.head()
dff.shape

(48842, 15)

Second choice: dataset 2

1. Source of data
- Kaggle

2. Brief description of data
- This data set shows an individual's income based on various factors.
3. What is the target?
income
4. What does one row represent? (A person? A business? An event? A product?)
- A person

5. Is this a classification or regression problem?
- Classification, as income is split between making more or less than 50k.

6. How many features does the data have?
- 15
7. How many rows are in the dataset?
- 48842

8. What, if any, challenges do you foresee in cleaning, exploring, or modeling this dataset?
- There are inconsistencies in the labelling of the education column that may take some time to really clean up.